pubsub/google/cloud/pubsub/topic.py

```diff
@@ -408,15 +410,40 @@ class Batch(object):
     :type topic: :class:`google.cloud.pubsub.topic.Topic`
     :param topic: the topic being published
 
-    :type client: :class:`google.cloud.pubsub.client.Client`
     :param client: The client to use.
+    :type client: :class:`google.cloud.pubsub.client.Client`
+
+    :param max_interval: The maximum interval, in seconds, before the batch
+                         will automatically commit. Note that this does not
+                         run a background loop; it just checks when each
+                         message is published. Therefore, this is intended
+                         for situations where messages are published at
+                         reasonably regular intervals. Defaults to infinity
+                         (off).
+    :type max_interval: float
+
+    :param max_messages: The maximum number of messages to hold in the batch
+                         before automatically commiting. Defaults to infinity
+                         (off).
+    :type max_messages: float
     """
-    def __init__(self, topic, client):
+    INFINITY = float('inf')
+
+    def __init__(self, topic, client, max_interval=INFINITY,
+                 max_messages=INFINITY):
         self.topic = topic
         self.messages = []
         self.message_ids = []
         self.client = client
 
+        # Set the autocommit rules. If the interval or number of messages
+        # is exceeded, then the .publish() method will imply a commit.
+        self._max_interval = max_interval
+        self._max_messages = max_messages
+
+        # Set the initial starting timestamp (used against the interval).
+        self._start_timestamp = float(time.time())
```

Comment:
Isn't it already a float?

---

`time.time()` isn't guaranteed to be monotonic, e.g. if there is a leap second. Python 3 does provide a monotonic clock

```python
$ python3.5 -c 'import time; print(time.time()); print(time.monotonic())'
1485491671.1016715
569331.546634608
```

but Python 2 does not

```python
$ python2.7 -c 'import time; print(time.time()); print(time.monotonic())'
1485491665.14
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'module' object has no attribute 'monotonic'
```